doc/api/util.md

```diff
@@ -771,9 +771,13 @@ added: v8.0.0
 * {symbol} that can be used to declare custom promisified variants of functions,
 see [Custom promisified functions][].
 
-## Class: util.TextDecoder
+## Class: TextDecoder
```

Comment:
I think this is confusing if `util.TextDecoder` is still accessible.

Reply:
@mscdex I did this for consistency since, in the `url` docs, the class heading is `URL` instead of `url.URL`: [https://nodejs.org/dist/latest-v10.x/docs/api/url.html#url_class_url](https://nodejs.org/dist/latest-v10.x/docs/api/url.html#url_class_url)

But I agree with you that this is confusing. In the case of `URL`, it fits the `url` docs. In the case of `TextDecoder`, it doesn't intrinsically fit the `util` docs.

So I'm gonna change this here, and also follow your suggestion of moving the `TextDecoder` and `TextEncoder` docs from `util` to `globals` (see: [https://github.com/nodejs/node/pull/20662#discussion_r187436043](https://github.com/nodejs/node/pull/20662#discussion_r187436043)).

Reply:
@mscdex Actually, I'm not quite sure about this. All the globals in `globals.md` link to other docs, so `globals` is, in effect, some kind of meta documentation. If I'd move the encoding docs to `globals`, 90% of the globals documentation would be about encoding, which might be confusing, too. So I don't think is a clear-cut issue.

In fact, I think there are 3 possibilities here:

**[1.]** Keep the encoding documentation in `util` and link to it from `globals`.

- `pro`: Keeps the old documentation structure so people that know the docs already won't be surprised.
- `con`: When searching for "encoding", "text encoding" or similar terms, people are less likely to find the documentation (since the words "globals" and "utilities" are not semantically related to "encoding").
 
**[2.]** Move the encoding documentation to `globals` and link to it from `util`.

- `con`: The `globals` docs are currently "meta documentation" in that they only link to other, more detailed documentation. Moving the encoding documentation into this section would break with that.
- `con`: As explained in `[1.]`, when searching for "encoding", less experienced people might not look in the "globals" documentation, making it less likely for them to find the correct documentation.
- `con (maybe)`: This could possibly make navigating the `globals` docs harder in the future, for example, if new globals are introduced that would occur below the lengthy encoding documentation.

**[3.]** Move the encoding documentation to a new documentation section, e.g. `encoding` / "Encoding" or `text-encoding` / "Text Encoding", and link to it from `util` and `globals`.

- `pro`: This would perhaps make finding the correct documentation easier for less experienced people.
- `con (maybe)`: Documentation-table-of-contents-bloat (if such a thing exists). 😄

In any case, I'm not sure what's a good way to proceed with this. So I'd greatly appreciate more feedback regarding this issue.

PS: When I go to [https://nodejs.org/dist/latest-v10.x/docs/api/](https://nodejs.org/dist/latest-v10.x/docs/api/), I sorely miss a tiny search-box in the top-right corner - I think that would help a lot.

Reply:
For overall doc search, you can use "View on single page" variant, there is a link at the top of each doc.

Reply:
@vsemozhetbyt True, but again, I don't think that's a friendly UX/DX. (Especially on mobile, loading the whole documentation on a single page might not be the best option. But then again, I think this is off-topic here.)